projects/testing-library/src/lib/models.ts

```diff
@@ -65,6 +65,12 @@ export interface RenderResult<ComponentType, WrapperType = ComponentType> extend
   ) => Promise<void>;
 }
 
+export interface ProviderInterface {
```

Comment:
Hi, thanks for creating this PR.

The reason why this is typed as `any` is because it's also like that within the Angular TestBed. Let me think on this for a while longer before deciding to merge this in.

Do you think this could potentially break something?

Reply:
Thanks for the feedback. Take your time. I know this is different from the Implementation in Angular and I have been thinking about raising an issue with them as well. 

So with all the types I looked at Angular proper (not the Testbed) and adapted accordingly. The `imports` type is straight from Angular proper. 

The `providers` types I basically looked at what does Angular proper have as type definition and made it a bit more loose. There is a `Provider` type in Angular, but intentionally I only tried to mimic the structure and not the types. `useValue` and `provide` are still `any`. I basically want to make it easier for people writing tests. Because as it is now you get no auto completion or even error if you make a mistake. 

That said I do not know if this could potentially break something.

Reply:
I opened an issue in the angular repo: https://github.com/angular/angular/issues/51765

Reply:
Apparently the Angular team is already at it: https://github.com/angular/angular/issues/37178

Reply:
Thanks @Christian24 , because the Angular team is busy with it I want to wait until they provide the proper types for it. Otherwise it could lead into breaking changes or other things.

Reply:
I agree. I subscribed myself to the PR. I will update mine as soon as it is merged.